R/party.R

```diff
@@ -0,0 +1,299 @@
+
+approx_surv_fit <- function(x, .time = 0) {
+  dat <- stack_survfit_cph(x, n = 1)
+  interpolate_km_values(dat, .time)
+}
+
+#' @export
+#' @keywords internal
+#' @rdname party_internal
+surv_prob_ctree <- function(object, new_data, .time) {
+  cl <-
+    rlang::call2(
+      "Predict",
+      .ns = "modeltools",
+      rlang::expr(object),
+      rlang::expr(new_data),
+      type = "prob"
+    )
+  res <- rlang::eval_tidy(cl)
+  res <- purrr::map(res, approx_surv_fit, .time)
+  res <- purrr::map(res, ~ dplyr::select(.x, -.pred_hazard_cumulative, -.row))
+  tibble::tibble(.pred = res)
+}
+
+# ------------------------------------------------------------------------------
+
+#' A wrapper function for conditional inference forests
+#'
+#' This function is a slightly different API for [party::cforest()] that has
+#' several important arguments as top-level arguments (as opposed to being
+#' specified in [party::cforest_control()]).
+#' @inheritParams cond_inference_surv_ctree
+#' @param ... Other options to pass to [party::cforest()].
+#' @param mtry number of input variables randomly sampled as candidates at each
+#' node for random forest like algorithms. The default \code{mtry = 0} means
+#' that no random selection takes place.
+#' @param replace a logical indicating whether sampling of observations is done
+#' with or without replacement.
+#' @param fraction fraction of number of observations to draw without
+#' replacement (only relevant if \code{replace = FALSE}).
+#' @param ntree number of trees to grow in a forest.
+#' @details
+#' Note that, although [party::cforest_unbiased()] is not directly used, the
+#' defaults for `cond_inference_forest()` mirror its default values. However,
+#' [party::cforest_unbiased()] does not allow several tuning parameters to be
+#' optimized (`teststat`, `testtype`, `mincriterion`, `replace`, and `fraction`).
+#' If you set pass a [party::cforest_unbiased()] object to
+#' `cond_inference_forest()` and modify those arguments, their values will be
+#' overwritten.
+#' @return An object of class `RandomForest-class`.
+#' @seealso cond_inference_surv_ctree
+#' library(survival)
+#' data(lung)
+#' cond_inference_surv_cforest(
+#'   Surv(time, status) ~ age + sex,
+#'   data = lung,
+#'   mtry = 1,
+#'   ntree = 10
+#' )
+#' @export
+cond_inference_surv_cforest <-
+  function(formula,
+           data,
+           minsplit = 20L,
+           maxdepth = 0L,
+           teststat = "quad",
+           testtype = "Univariate",
+           mincriterion = 0,
+           replace = FALSE,
+           fraction = 0.632,
+           mtry = 5L,
+           ntree = 500L,
+           ...) {
+    opts <- rlang::list2(...)
+
+    # Edit the control options to put some argument values into the controls
+    # function. This is a challenge since the control objects are S4 objects
+    # of S4 objects. If the top-level tuning parameter disagrees with an
+    # existing value, we over-write it an issue a warning.
```

Comment:
```suggestion
    # existing value, we over-write it and issue a warning.
```

Reply:
for the cases below it's `warn = FALSE` whereas for the ctree function warnings get issued.should the two fit wrappers have the same behaviour re: warnings?